### PR TITLE
Make DirectDraw optional

### PR DESCRIPTION
**Pull request by wdigger**
Main idea - encapsulate platform (framework) dependent code in separated library.
Next step - to do the same with audio, input and event circle.
Like a proof of concept I plan to use SDL2 as backend.

### CODE REVIEW

**Comment by hifi** [comment]
I have no issues with this approach. Just try to keep the commits small and incremental where possible.

**Comment by wdigger** [comment]
Unfortunately, right now project contains too much unnecessary includes, even small change in header may be a casualty of massive changes in whole project. I believe we will cleanup headers in some point.

All those changes are result of simple action - remove all <ddraw.h> and "ddraw.h" to only one source file.

**Comment by hifi** [comment]
I'm really interested in getting this merged so that the remaster build would be ddraw free as gbuffer is one of the biggest obstacles in getting things to compile on Linux at all (many files depend on the header).

And I mean in a way that first we would merge a solution (this PR) that separates the drawing stuff from remaster builds so the first portable builds are the remaster game libraries even when they are completely useless without the remaster. That allows incremental portability efforts and we can work on fixing video and event system later on in separate PRs. First there's a lot of winapi cleanup that needs to be done even with the smaller remaster build.

**Review by hifi — CHANGES_REQUESTED** [review]
RA crashes for me before it gets to main menu with null video. It shouldn't draw anything for sure but it shouldn't crash either. I couldn't get a good stack trace from winedbg.

Otherwise I think these changes are good as without ddraw the ddraw.h header is not required to build and remaster seems to work fine.

**Comment by wdigger** [comment]
RA crash fixed

File: common/video_null.cpp [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+ *                  int bits_per_pixel  - the number of bits per pixel the video mode supports *
+ *                                                                                             *
+ * OUTPUT:     none                                                                            *
+ *                                                                                             *
+ * HISTORY:                                                                                    *
+ *   09/26/1995 PWG : Created.                                                                 *
+ *=============================================================================================*/
+bool Set_Video_Mode(int w, int h, int bits_per_pixel)
+{
+    return false;
```
Comment (hifi) [inline common/video_null.cpp]: This should be true as the null video driver should "work".

Setting it to true still crashes RA. The game should run without rendering.

**Review by hifi — CHANGES_REQUESTED** [review]

File: common/video_null.cpp (hunk 1 above)
Comment (wdigger) [reply common/video_null.cpp]: Done. But, without graphics, it's not clear what application actually do. At least no crashes.

**Review by hifi — APPROVED** [review]
Tested all combinations and all seemed to work just fine with ddraw and null video. I could even navigate into the game from the menu with the keyboard with dsound enabled so I knew the game was running.

Excellent work, thanks!

**Pull request merged**